lib/config/config.js

```diff
@@ -29,6 +32,35 @@ var isInitialized = false;
 loggingService.addWriter('terminal', ['>=notice'], {});
 
 /**
+ * ConfigurationError class
+ */
+class ConfigurationError extends Error {
+	constructor(message, filename, error) {
+		super(message);
+
+		this.name = 'ConfigurationError';
+		this.filename = filename;
```

Comment:
JS `Error` class [may already](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error) expose a `fileName` property. To avoid confusion, you may want to rename this.

Reply:
Any suggestions?

Reply:
I will merge it as-is for now; this is a Mozilla-specific error field, and I don't think it is something we can expect to see change anytime soon. And I fear using a different name may not be clear enough.